src/DittoMachine.sol

```diff
@@ -372,8 +388,12 @@ contract DittoMachine is ERC721, ERC721TokenReceiver, ERC1155TokenReceiver, Clon
             return MIN_AMOUNT_FOR_NEW_CLONE;
         }
         CloneShape memory cloneShape = cloneIdToShape[cloneId];
-        uint256 _minAmount = _getMinAmount(cloneShape);
-        return _minAmount + (_minAmount * MIN_FEE * (1 + cloneShape.heat) / DNOM);
+        bool intraBlock = _getBlockRefund(cloneId) != 0;
+        uint256 _minAmount = _getMinAmount(cloneShape, intraBlock);
+        // calculate inverse percentage of fee amount
+        uint256 feePercent = _minAmount * (MIN_FEE * ((intraBlock ? 0:1) + cloneShape.heat));
+        uint256 feePortion = (feePercent * DNOM) / (DNOM - (MIN_FEE * ((intraBlock ? 0:1) + cloneShape.heat))) / DNOM;
```

Comment:
i think these lines are complicated to understand for the reader. can you simplify this a bit. for example, by extracting `(MIN_FEE * ((intraBlock ? 0:1) + cloneShape.heat))` in a variable?

also shouldn't the return value be `_minAmount + feePercent/DNOM`?

Reply:
yes I can simplify into another variable.

No the return value needs to be a new calculated amount based on the minimum fee. Doing a kind of reverse percentage calculation here. Because the fee is calculated on the amount transferred now, the percentages are not the same at higher values, so the fee needs to be calculated as part of the minimum transferred amount.

Does that make sense?

I may have explained this poorly and finding it kind of difficult through text. I can explain further if I wasn't clear enough.